### PR TITLE
Linode apiv4 support

### LINKED ISSUE #56319: salt-cloud doesn't support linode v4 API (and it's now impossible to get v3 API keys)
We were using salt-cloud to provision servers on Linode, and all was good, until our API key expired. :-(

It appears the old Linode API is still available and working (we have some valid API keys on our dev Linode account which appear to be working), but the old manager was decommissioned in January 2020, so you can't make new API keys for the old API:

https://www.linode.com/2019/11/18/notice-legacy-linode-manager-users/

I looked at the code in the master branch for the Linode driver and it looks like it's all specific to the old API (you can tell from the all caps JSON keys):

https://github.com/saltstack/salt/blob/master/salt/cloud/clouds/linode.py

Going forward, salt-cloud is going to need to interact with the API, because no one is going to be able to get keys for the old API (and probably at some point they'll decommission the old API too). m. Until that point, we're going to be locked out from using salt-cloud!

### PR DESCRIPTION
**Pull request by 0xch4z**
### What does this PR do?

This PR adds opt-in Linode APIv4 support to the Linode cloud module via the `api_version` cloud configuration parameter.

### What issues does this PR fix or reference?
Fixes: #56319
### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [X] Docs
- [x] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [X] Tests written/updated

### CODE REVIEW

**Comment by garethgreenaway** [comment]
re-run ubuntu

**Comment by garethgreenaway** [comment]
re-run windows

**Comment by garethgreenaway** [comment]
re-run docs

**Comment by Charliekenney23** [comment]
@dwoz Can you take another look when you get a sec 😃

It seems like the build is detecting multiple changelog entries and that's why the lint stage is failing. But all tests are passing now

**Comment by bryceml** [comment]
We should actually run this against the cloudtests, doing that here: https://github.com/saltstack/salt/pull/58248

File: salt/cloud/clouds/linode.py [hunk 1]
```diff
@@ -1,38 +1,118 @@
...
+
+Profile
+-------
+
+The following profile parameters are supported:
+
+- **size**: (required) The size of the VM. This should be a Linode instance type ID (i.e. ``g6-standard-2``). For APIv3, this would be a plan ID (i.e. ``Linode 2GB``). Run ``salt-cloud -f avail_sizes my-linode-provider`` for options.
+- **location**: (required) The location of the VM. This should be a Linode region (e.g. ``us-east``). For APIv3, this would be a datacenter location (i.e. ``Newark, NJ, USA``). Run ``salt-cloud -f avail_locations my-linode-provider`` for options.
+- **image**: (required) The image to deploy the boot disk from. This should be an image ID (e.g. ``linode/ubuntu16.04``); official images start with ``linode/``. For APIv3, this would be an image label (i.e. Ubuntu 16.04). Run ``salt-cloud -f avail_images my-linode-provider`` for more options.
+- **password**: (\*required) The default password for the VM. Must be provided at the profile or provider level.
```
Comment (bryceml) [inline salt/cloud/clouds/linode.py:53]: what's with the `\*required` ?

Comment (Charliekenney23) [reply salt/cloud/clouds/linode.py:53]: > Must be provided at the profile or provider level.
I was trying to convey that a `password` value is required, but not necessarily at the VM configuration level.

Comment (bryceml) [reply salt/cloud/clouds/linode.py:53]: I guess that works.

**Comment by bryceml** [comment]
Also, all of the class docs show up in the html docs which makes them not too pretty.  I think there's a flag in sphinx to hide them, but I can't remember what it is.

**Comment by bryceml** [comment]
no linode cloud test failures, so we're good there.
```
Test Result (8 failures )

    integration.cloud.clouds.test_ec2.EC2Test.test_instance
    integration.cloud.clouds.test_ec2.EC2Test.test_instance_rename
    integration.cloud.clouds.test_ec2.EC2Test.test_win2012r2_psexec
    integration.cloud.clouds.test_ec2.EC2Test.test_win2012r2_winrm
    integration.cloud.clouds.test_ec2.EC2Test.test_win2016_psexec
    integration.cloud.clouds.test_ec2.EC2Test.test_win2016_winrm
    integration.cloud.clouds.test_gce.GCETest.test_instance_extra
    integration.cloud.clouds.test_msazure.AzureTest.test_instance
```
which is the same as the master branch currently.

**Comment by bryceml** [comment]
something like 
```
diff --git a/doc/ref/clouds/all/salt.cloud.clouds.linode.rst b/doc/ref/clouds/all/salt.cloud.clouds.linode.rst
index 30f987dcbc..d5b0c23411 100644
--- a/doc/ref/clouds/all/salt.cloud.clouds.linode.rst
+++ b/doc/ref/clouds/all/salt.cloud.clouds.linode.rst
@@ -3,4 +3,5 @@ salt.cloud.clouds.linode
 ========================
 
 .. automodule:: salt.cloud.clouds.linode
-    :members:
\ No newline at end of file
+    :members:
+    :exclude-members: LinodeAPI, LinodeAPIv3, LinodeAPIv4
```
should fix up the docs

**Comment by Charliekenney23** [comment]
@bryceml all fixed!

**Review by bryceml — APPROVED** [review]

**Comment by Akm0d** [comment]
If the lint failures are unrelated to your changes then you just need to periodically update your branch until the unrelated failures are fixed.  Even if the failures are unrelated PRs won't get merged until all the tests are green :)

**Review by Akm0d — APPROVED** [review]

**Comment by Charliekenney23** [comment]
@Akm0d @sagetherage @bryceml Anything I can do to get this merged?

**Comment by sagetherage** [comment]
I believe we have been working to fix test failures (outside of this PR) before merging new code, so it seems we are delayed or forgetful, however the team is focused on fixing test suite issues so merging is smoother in the near future. That being said, I will see about this today and early next week to get it merged. Thank you for the mention!

**Comment by Charliekenney23** [comment]
Thanks, everyone for getting this merged in!!

**Pull request merged**